### PR TITLE
Add feed pagination.

### PR DESCRIPTION
**Pull request by kimlai**
- Create a FeedStore that contains the `nextLink` for pagination, and
  the ordered list of track ids.
- The TrackStore now only contains a Map of tracks.
- The server needs some refactoring for DRYing things up.

### CODE REVIEW

File: modules/tracks/stores/FeedStore.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+var toImmutable = require('nuclear-js').toImmutable;
+var Store = require('nuclear-js').Store;
+var RECEIVE_FEED = require('../actionTypes').RECEIVE_FEED;
+
+module.exports = new Store({
+    getInitialState: function () {
+        return toImmutable({ tracks: [] });
```
Comment (Sinewyk) [inline modules/tracks/stores/FeedStore.js]: Any reason why not simply toImmutable([]) ?

Comment (kimlai) [inline modules/tracks/stores/FeedStore.js]: `Uncaught TypeError: Cannot read property 'concat' of undefined`

From the `concat` call in `receiveFeed`.

Comment (kimlai) [inline modules/tracks/stores/FeedStore.js]: And I'm using an object because the state also contains a `nextLink` property.

Comment (kimlai) [inline modules/tracks/stores/FeedStore.js]: And I'm using a `nextLink` because that's the way soundcloud does pagination.

Comment (kimlai) [inline modules/tracks/stores/FeedStore.js]: Added a null nextLink the the initial state for clarity.

**Pull request merged**